### DISCUSSION TITLE
Download .txt file in Wave

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1560] [Q&A] [answered]
Hi team,

I am facing a problem when I was trying to download a text file using ui.link component. Sample test code parts are as below. 

Write the file and upload it to the server,
```py
    users_list_file_path = os.path.join(q.app.users_list_dir, 'users_admins.txt')

    if os.path.exists(users_list_file_path):
        os.remove(users_list_file_path)

    users_list_file = open(users_list_file_path, "a")

    users_list_file.write(f"Users - {len(users)}\n\n")
    write_export_users_file_content(users, users_list_file)

    users_list_file.write(f"\nAdmins - {len(admins)}\n\n")
    write_export_users_file_content(admins, users_list_file)

    users_list_file.close()
    (q.app.users_list_file_link,) = await q.site.upload([users_list_file_path])
```
Component file code line,
```py
                        ui.link(label="↗ Export Users", path=q.app.users_list_file_link,
                                button=True, download=True, target='_blank'),
```
This opens the new browser tab with the content but not downloading. When I tested with the file extension .sig, .docx it downloaded the file with relevant extension. 

Thanks in advance!

**Comment by aalencar** [comment, marked as answer]
> This opens the new browser tab with the content but not downloading. When I tested with the file extension .sig, .docx it downloaded the file with relevant extension.

I haven't tested but maybe the browser tries to display `.txt` files and downloads the ones it doesn't support, like `.docx`? What happens if you remove `target=_blank`?

**Reply by shamilh2o** [reply]
Hi, Thanks for the suggestion.
I tried both removing target and keeping target as an empty string. But still for .txt it displays content in a new tab without downloading it. 

cc: @mturoci

**Reply by mturoci** [reply]
@aalencar is right - https://github.com/h2oai/wave/issues/999#issuecomment-1153710468

**Reply by shamilh2o** [reply]
Got it. Thanks @mturoci
